disnake/channel.py

```diff
@@ -448,13 +449,83 @@ async def edit(
             # the payload will always be the proper channel payload
             return self.__class__(state=self._state, guild=self.guild, data=payload)  # type: ignore
 
-    @utils.copy_doc(disnake.abc.GuildChannel.clone)
     async def clone(
-        self, *, name: Optional[str] = None, reason: Optional[str] = None
+        self,
+        *,
+        name: Optional[str] = None,
+        topic: Optional[str] = MISSING,
+        position: Optional[int] = MISSING,
+        nsfw: Optional[bool] = MISSING,
+        category: Optional[Snowflake] = MISSING,
+        slowmode_delay: Optional[int] = MISSING,
+        type: Optional[ChannelType] = MISSING,
+        default_auto_archive_duration: Optional[AnyThreadArchiveDuration] = MISSING,
+        flags: Optional[ChannelFlags] = MISSING,
+        reason: Optional[str] = None,
```

Comment:
```suggestion
        name: Optional[str] = None,
        topic: Optional[str] = MISSING,
        position: int = MISSING,
        nsfw: bool = MISSING,
        category: Optional[Snowflake] = MISSING,
        slowmode_delay: int = MISSING,
        type: Optional[ChannelType] = MISSING,
        default_auto_archive_duration: AnyThreadArchiveDuration = MISSING,
        flags: Optional[ChannelFlags] = MISSING,
        reason: Optional[str] = None,
```

These are a little bit inconsistent with the other methods; there's a `position: int` in one place and `position: Optional[int]` in another.

On the other hand, there may be an argument to be made for all parameters to be optional, so that different behaviors can be achieved:
- `clone(attr=123)` -> sets attr to `123`
- `clone(attr=None)` -> sends `attr: null`, which results in the default value being used ***
- `clone()` -> uses existing channel's attr

@onerandomusername (and others, of course), thoughts?

Reply:
> * `clone(attr=None)` -> sends `attr: null`, which results in the default value being used ***

What do you mean by "default value", the default value that discord uses, right?

I like the idea though 👍

Reply:
> What do you mean by "default value", the default value that discord uses, right?
>
> I like the idea though :+1:

~~it's been two months, I don't quite remember what I had in mind here :thonk:~~

Yes, it would be equivalent to not sending the field on channel creation, and letting Discord assign their default value for that field.

Thinking about it again, I think this wouldn't be worth the additional complexity. If you wanted to e.g. use the default `user_limit`, you'd just use `user_limit=0`; there's no real advantage in being able to use `user_limit=None`.

The first part of my original comment seems to be resolved now, ignore the rest talking about making everything optional :sweat_smile:

Reply:
I'm marking this conversation as solved